services/ui/src/lib/components/VariableDiag/Histogram.svelte

```diff
@@ -0,0 +1,87 @@
+<script>
+  import { format, scaleLinear, min, max, extent } from "d3";
+
+  export let width = 0;
+  export let height = 0;
+
+  export let bins = [];
+
+  let margin = {
+    top: 0,
+    right: 0,
+    bottom: 32,
+    left: 0,
+  };
+
+  let formatNum = format("~s");
+
+  $: domain = [min(bins, (d) => d.lower), max(bins, (d) => d.upper)];
```

Comment:
Today I learned about reactive statements, and they're really cool. https://geoffrich.net/posts/svelte-$-meanings/#heading-reactive-statements

Reply:
Yeah. This is also a Svelte-specific thing (which actually uses real JavaScript syntax, but does not behave the same way, which I think is a little confusing).